File: src/main/java/io/github/kingg22/api/vacunas/panama/persistence/repository/DistritoRepository.java
package io.github.kingg22.api.vacunas.panama.persistence.repository;

import io.github.kingg22.api.vacunas.panama.persistence.entity.Distrito;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface DistritoRepository extends JpaRepository<Distrito, Short> {

    Optional<Distrito> findDistritoByNombre(String nombre);

    Optional<Distrito> findDistritoByNombreAndProvincia_Id(String nombre, Short idProvincia);

    Optional<Distrito> findDistritoByNombreAndProvincia_Nombre(String nombre, String nombreProvincia);
}

File: src/main/kotlin/io/github/kingg22/api/vacunas/panama/modules/direccion/repository/DistritoRepository.kt
package io.github.kingg22.api.vacunas.panama.modules.direccion.repository

import io.github.kingg22.api.vacunas.panama.modules.direccion.entity.Distrito
import io.github.kingg22.api.vacunas.panama.util.withSession
import io.quarkus.hibernate.reactive.panache.kotlin.PanacheRepositoryBase
import io.smallrye.mutiny.coroutines.awaitSuspending
import jakarta.enterprise.context.ApplicationScoped

@ApplicationScoped
class DistritoRepository : PanacheRepositoryBase<Distrito, Short> {
    suspend fun findByIdOrNull(id: Short): Distrito? = withSession { findById(id).awaitSuspending() }
    suspend fun listAllDistritos() = withSession { listAll().awaitSuspending() }
}
